src/lua_console.cpp

```diff
@@ -0,0 +1,82 @@
+#include "lua_console.h"
+
+#include "catalua.h"
+#include "catacharset.h"
+
+lua_console::lua_console() : cWin( newwin( lines, width, 0, 0 ) ), iWin( newwin( 1, width,lines, 0 ) )
+{
+}
+
+void lua_console::dispose()
+{
+    werase( cWin );
+    werase( iWin );
+    delwin( cWin );
+    delwin( iWin );
+}
+
+std::string lua_console::get_input()
+{
+    long key = 0;
+    int pos = -1;
+    return string_input_win( iWin, "", width, 0, 0, width, true, key, pos, "LUA", 0, lines, true, false, callbacks );
+}
+
+void lua_console::draw()
+{
+    werase( cWin );
+
+    // Some juggling to make sure text is aligned with the bottom of the console.
+    int stack_size = text_stack.size() - scroll;
```

Comment:
I like the idea, but putting all the text into one string would break the scrolling functionality. We would have to cut up the string again by newlines to know from where to start, which wouldn't work, because the strings that end up being too long and then folded, wouldn't have newlines in them.

And I wouldn't know how many lines the text is going to be before calling ```fold_and_print```, so I couldn't keep it aligned with the bottom of the console.

I do have a compromise though, I'll commit it.